### ISSUE TITLE
tangential position range inconsistency

### ISSUE THREAD

**Issue by SomayehSaghamanesh** [UCL/STIR#1507]
There are two different definitions (as below), which result in inconsistency in tangential position range. This is problematic at least in cases handling full tangential positions.
In ProjDataInfo.cxx:
  min_tangential_pos_num = -(num_tang_poss / 2);
  max_tangential_pos_num = min_tangential_pos_num + num_tang_poss - 1;

In ProjDataInfoCylindricalNoArcCorr.cxx and ProjDataInfoGenericNoArcCorr.cxx :
  const int min_tang_pos_num = -(num_detectors / 2) + 1;
  const int max_tang_pos_num = -(num_detectors / 2) + num_detectors;

**Comment by robbietuk** [comment]
The class inheritance is as follows
![image](https://github.com/user-attachments/assets/bc2a34a7-78fc-43e0-9c56-523c3fe7cf4d)

I suggest using this https://github.com/UCL/STIR/blob/e168b074b3f7d21b4ebf8872680b4bb115022719/src/buildblock/ProjDataInfo.cxx#L123-L129
in the child classes for consistency?

Maybe worth noting, the definition of the max/min tang poss in `ProjDataInfoCylindricalNoArcCorr` is very old (circa 2003). Somethings may break because they were built upon this methodology.

**Comment by SomayehSaghamanesh** [comment]
> The class inheritance is as follows ![image](https://private-user-images.githubusercontent.com/13292409/366146110-bc2a34a7-78fc-43e0-9c56-523c3fe7cf4d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwNDEzMDcsIm5iZiI6MTcyNjA0MTAwNywicGF0aCI6Ii8xMzI5MjQwOS8zNjYxNDYxMTAtYmMyYTM0YTctNzhmYy00M2UwLTljNTYtNTIzYzNmZTdjZjRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTExVDA3NTAwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRhOTQ5MGZmZjUyZWNiYzM1NWNkZTM1YTIxOTBjZDM4OGE5ODJkNGM0MWI5ZjAyYmU4ODEyZmZlYjBjNzU5ZmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cMah34pBBbg9EUXLPkJEL5HIp56DK63z-gj-oW86XKg)
> 
> I suggest using this
> 
> https://github.com/UCL/STIR/blob/e168b074b3f7d21b4ebf8872680b4bb115022719/src/buildblock/ProjDataInfo.cxx#L123-L129
> 
> in the child classes for consistency?
> 
> Maybe worth noting, the definition of the max/min tang poss in `ProjDataInfoCylindricalNoArcCorr` is very old (circa 2003). Somethings may break because they were built upon this methodology.

Yes, I have already followed it in PR #1508, changed two classes to be consistent with ProjDataInfo.cxx.

**Comment by robbietuk** [comment]
I am sorry, I reread the full function. These values are used for slightly odd checks in `initialise_uncompressed_view_tangpos_to_det1det2`. I'm not sure what these https://github.com/UCL/STIR/blob/e168b074b3f7d21b4ebf8872680b4bb115022719/src/buildblock/ProjDataInfoGenericNoArcCorr.cxx#L152-L163 checks are, but since both classes' methods are a copy and paste of one another. Your fix in #1508 seems correct to me.

**Comment by SomayehSaghamanesh** [comment]
Just not sure why 8 tests (including FBP, OSMAPOSL, etc) are failed after these changes. Haven't gone through them yet. Any comment is appreciated.
